modules/apps/static/osgi/osgi-util/src/main/java/com/liferay/osgi/util/service/Snapshot.java

```diff
@@ -89,6 +91,18 @@ public Snapshot(
 
 				serviceTracker.open();
 
+				bundleContext.addBundleListener(
+					new SynchronousBundleListener() {
+
+						@Override
+						public void bundleChanged(BundleEvent bundleEvent) {
+							serviceTrackerDCLSingleton.destroy(null);
```

Comment:
@tinatian this can not be right, it is not even checking the changed bundle is the one you got the previous BundleContext or not. We will just end up with tons of meaningless destroying during startup, since there are thousands bundles are changing there states.